app/src/main/java/com/github/multimatum_team/multimatum/activity/AddDeadlineActivity.kt

```diff
@@ -412,11 +468,7 @@ class AddDeadlineActivity : AppCompatActivity() {
     }
 
     // Recuperate the information on which notification must be set before returning it in an array
-    private fun retrieveNotificationsTimes(): ArrayList<Long> {
-        val res = ArrayList<Long>()
-        for (i in 0 until checkBoxIdTime.count()) {
-            if (notificationSelected[i]) res.add(checkBoxIdTime[nameNotifications[i]]!!)
-        }
-        return res
+    private fun retrieveNotificationsTimes(): List<Long> {
+        return timeNotifications.filterIndexed { idx, _ -> notificationSelected[idx] }
     }
```

Comment:
nice improvment of readability

Reply:
Yes, it's all thanks to Valentin who has showedme filterIndexed